.github/workflows/build.yml

```diff
@@ -0,0 +1,338 @@
+          echo "::set-output name=go-version::$(cat .go-version)"
+  get-product-version:
+    runs-on: ubuntu-latest
+    outputs:
+      product-version: ${{ steps.get-product-version.outputs.product-version }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: get product version
+        id: get-product-version
+        run: |
+          make version
+          echo "::set-output name=product-version::$(make version)"
+  generate-metadata-file:
+    needs: get-product-version
+    runs-on: ubuntu-latest
+    outputs:
+      filepath: ${{ steps.generate-metadata-file.outputs.filepath }}
+    steps:
+      - name: 'Checkout directory'
+        uses: actions/checkout@v2
+      - name: Generate metadata file
+        id: generate-metadata-file
+        uses: hashicorp/actions-generate-metadata@v1
+        with:
+          version: ${{ needs.get-product-version.outputs.product-version }}
+          product: ${{ env.PKG_NAME }}
+          repositoryOwner: "hashicorp"
+      - uses: actions/upload-artifact@v2
+        with:
+          name: metadata.json
+          path: ${{ steps.generate-metadata-file.outputs.filepath }}
+  set-ld-flags:
+    needs: get-product-version
+    runs-on: ubuntu-latest
+    outputs:
+      ldflags: ${{ steps.generate-ld-flags.outputs.ldflags }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: 'Generate ld flags'
+        id: generate-ld-flags
+        run: |
+          project="$(go list -m)"
+          sha="$(git rev-parse --short HEAD)"
+          echo "::set-output name=ldflags::"-s -w -X \'$project/version.Name=${{ env.PKG_NAME }}\' \
+          -X \'$project/version.GitCommit=$sha\' \
+          -X \'$project/version.GitDescribe=v$(make version base=1)\'""
+  
+  build-other:
+    needs: [ get-go-version, get-product-version, set-ld-flags ]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        goos: [ windows ]
+        goarch: [ "386", "amd64" ]
+      fail-fast: true
+
+    name: Go ${{ needs.get-go-version.outputs.go-version }} ${{ matrix.goos }} ${{ matrix.goarch }} build
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ needs.get-go-version.outputs.go-version }}
+    
+      - name: Build dependencies
+        run: |
+          make deps
+      
+      - name: Build generate-all
+        run: make generate-all
+      
+      - name: Setup node and yarn
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+          cache: 'yarn'
+          cache-dependency-path: 'ui/yarn.lock'
+      
+      - name: Install Yarn
+        run: |
+          npm install -g yarn
+      
+      - name: UI Build
+        run: |
+          cd ui
+          yarn install --network-timeout 300000
+          yarn --verbose run build
+          cd ..
+      
+      - name: Build static assets
+        run: make static-assets  
+
+      - name: Build
```

Comment:
Before we build, we [make a commit](https://github.com/hashicorp/nomad/blob/main/scripts/release/Makefile.linux#L23-L32) with the generated files so that the release binary points to a git SHA that has them already. This guarantees that builds from the same SHA are always the same.

Reply:
I can definitely take a look into how we can add this!

Reply:
This is interesting. With CRT we're following a CD model, where builds are triggered on every merge to `main`/release branches. Should this automation be triggered at merge time before the build happens, or more frequently?

Reply:
Hum...not sure if I followed. We create these files before the build and only at release time, so not that often.

Reply:
Something to note is that not having these files in the release builds will introduce a fairly serious performance regression, but they're also slow and expensive to build (and would balloon changesets), which is why we don't do it on every commit today.